### PR TITLE
test: distinguish unit, integration and e2e tests

### LINKED ISSUE #263: Distinguishing different types of tests
As a Code PushUp contributor, I want to be able to add and view test files and be able to tell at the first glance what their scope is.

This issue is a part of #145 

## Implementation details
- [x] Each type of test has a separate `nx` target and runs in a separate CI pipeline job for every major OS.
- [x] Add an ESLint rule to enforce the new naming.
- [x] Each type of test has a separate `vite` config.
- [x] Rename `tsconfig.spec` to `tsconfig.test` and include all test patterns in it.
- [x] Adjust coverage pipeline for unit and integration tests.

### E2E tests
- Test the full flow in a real environment.
- `nx` target `test-e2e`
- Live in a separate `e2e` folder.
- E2E test files have the `.e2e.test.ts` suffix.

### Integration tests
- Test a bigger unit (e.g. a plugin) or an integration with a 3rd party tooling (e.g. `multi-progress-bars`).
- `nx` target `test-integration`
- Live next to their implementation files.
- Have the `.integration.test.ts` suffix.

### Unit tests
- Test a small unit (a function) which can be run in an isolated environment.
- `nx` target `test-unit`
- Live next to their implementation files.
- Have the `.unit.test.ts` suffix.

### PR DESCRIPTION
**Pull request by Tlacenka**
Closes #263 

In this PR, I did the following:
- I split the `test` target for unit and integration tests - the new test file format is `.unit.test.ts`, `.integration.test.ts` and `.e2e.test.ts`.
   - I used my feedback from the test refactor issues to determine which scope each file should have. This is of course up for discussion.
   - In `testing-utils`, I do not expect any integration or E2E tests so I only set up unit tests.
- I split the `vitest` config for different types of tests.
- I updated ESLint to accept the new file formats.
- I created defaults for the new `nx` targets and included all duplicate options in `nx.json`.
- I updated our CI to distinguish between different types of tests both when running checks and when uploading the code coverage (verified in [temporary PR](https://github.com/code-pushup/cli/pull/292) and [codecov](https://app.codecov.io/gh/code-pushup/cli/tree/codecov-check)).

I split the commits based on area or package for an easier review.

### CODE REVIEW

File: nx.json [hunk 1]
```diff
@@ -63,12 +84,12 @@
...
       "!{projectRoot}/.eslintrc.json",
-      "!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
-      "!{projectRoot}/tsconfig.spec.json",
+      "!{projectRoot}/**/?(*.)test.[jt]s?(x)?(.snap)",
+      "!{projectRoot}/tsconfig.test.json",
       "!{projectRoot}/src/test-setup.[jt]s",
       "!{projectRoot}/test-setup.[jt]s",
       "!{projectRoot}/**/?(*.)mock.[jt]s?(x)",
-      "!{projectRoot}/vite.config.[jt]s",
+      "!{projectRoot}/vite.config.@(unit|integration).[jt]s",
```
Comment (matejchalk) [inline nx.json:92]:
```suggestion
      "!{projectRoot}/vite.config.@(unit|integration|e2e).[jt]s",
```

File: packages/cli/project.json [hunk 2]
```diff
@@ -37,12 +37,18 @@
...
-        "passWithNoTests": true,
-        "reportsDirectory": "../../coverage/cli"
+        "reportsDirectory": "../../coverage/cli/unit"
+      }
+    },
+    "integration-test": {
+      "executor": "@nx/vite:test",
+      "outputs": ["{options.reportsDirectory}"],
+      "options": {
+        "reportsDirectory": "../../coverage/cli/integration"
```
Comment (matejchalk) [inline packages/cli/project.json:51]: Maybe explicit folder names would be clearer?

```suggestion
        "reportsDirectory": "../../coverage/cli/unit-tests"
      }
    },
    "integration-test": {
      "executor": "@nx/vite:test",
      "outputs": ["{options.reportsDirectory}"],
      "options": {
        "reportsDirectory": "../../coverage/cli/integration-tests"
```

(Would apply to all `project.json`s, obviously.)

**Review by matejchalk — APPROVED** [review]
Looking good :+1:

File: packages/cli/project.json (hunk 2 above)
Comment (Tlacenka) [reply packages/cli/project.json]: Makes sense. Done!

File: nx.json (hunk 1 above)
Comment (Tlacenka) [reply nx.json]: Good point!

**Review by matejchalk — APPROVED** [review]

**Pull request merged**